forge/forge/op/loss.py

```diff
@@ -119,3 +119,18 @@ def forward(self, predictions, labels):
         square = Multiply("square", diff, diff)
         loss = reduce_loss(self.reduction, square)
         return loss
+
+
+class NLLLoss(ForgeModule):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.is_loss = True
+
+    @validate_shapes(min_dim=2, max_dim=2)
+    def forward(self, prediction, labels):
+        weighted_prediction = Multiply("mul_pred_labels", prediction, labels)
+        rows = ReduceSum("r_sum", weighted_prediction, -1)
+        loss = ReduceAvg("r_avg", rows, -2)
```

Comment:
nit: can we move this at the end after Multiply and use dim=0 instead of -2. Just to align with CrossEntroyLoss since those are similar